src/gui/mzroll/tabledockwidget.cpp

```diff
@@ -2424,16 +2425,26 @@ PeakTableDockWidget::PeakTableDockWidget(MainWindow *mw) : TableDockWidget(mw) {
   toolBar->addAction(btnX);
 
   setTitleBarWidget(toolBar);
+
+  deletionDialog = new PeakTableDeletionDialog(this);
+  connect(deletionDialog, SIGNAL(accepted()),
+          this, SLOT(destroy()));
+  connect(deletionDialog, SIGNAL(rejected()),
+          deletionDialog, SLOT(hide()));
```

Comment:
>Maybe the "rejected" signal can be connected internally within PeakDetectionDialog, but "accepted" needs to be connected with the "destroy" slot of an instance of PeakTableDockWidget.

Well, the "accpeted" signal can still be connected to `PeakTableDockWidget::destroy()`  from `PeakTableDeletionDialog` class because  `PeakTableDockWidget` class is the parent of `PeakTableDeletionDialog` . 


> So I connected both the response signals in the same place -- looks better that way. I can of course move the "accepted" connection if you think it makes more sense to not make it explicit.

I suggested this because, to me  it makes more sense to have all the connections of a class defined in the ctor of that class. It makes easier to grep the code base. So, if I am looking for the all connections of a class I just have to check the ctor of that class. 
What do you think ? Maybe we should have some sort of **coding conventions** where we can define such things

Reply:
Right! So while we do not have any coding conventions around this, I think your way makes sense. I had forgot that the dialog's parent is passed as a param, and is accessible from within. I'll make changes accordingly.